sdk/azcore/arm/runtime/poller.go

```diff
@@ -22,26 +22,41 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
 )
 
+// NewPollerOptions contains the optional parameters for NewPoller.
+type NewPollerOptions[T any] struct {
+	// FinalState contains the final-state-via value for the LRO.
+	FinalState string
```

Comment:
@chlowell rename to `FinalStateVia`?

Reply:
Sure, that lines up better with the schema. I take it this is too obscure to warrant an enum?

Reply:
Hmm we could add an enum.  Per the docs, there is only a predefined set of values.

Reply:
Also, it looks like this might not be just for ARM.  I'm following up on that.

Reply:
Confirmed as I see it in some CS specs.  Will update.

Reply:
Actually I'll do that separately as the data-plane pollers need to be updated to support it.

Reply:
Opened https://github.com/Azure/azure-sdk-for-go/issues/17415 to track it.